File: ruby/rna-transcription/rna_transcription.rb

module BookKeeping   
    VERSION = 4
  end
  
  
  class Complement
    
    def self.of_dna(sequence)
  
      rna = sequence
              .split("")
              .map { |x| case x
                          when "G" then "C"
                          when "C" then "G"
                          when "T" then "A"
                          when "A" then "U"
                          else "" 
                        end }
              .join("")
      
      rna.size == sequence.size ? rna : ""
      
    end       
                       
  end 

File: javascript/rna-transcription/rna-transcription.js
var DnaTranscriber = function() {
 
};

DnaTranscriber.prototype.toRna = function(dna) {
  let rna = dna.split('').map(
        x =>
        x == 'G' ? 'C' : 
        x == 'C' ? 'G' : 
        x == 'T' ? 'A' : 
        x == 'A' ? 'U' : ''
    )
    .join('');

  if (!dna.length || !rna.length) {
    throw new Error('Invalid input'); 
  } else if (rna.length != dna.length  ) {
    throw new Error;
  } else {
  return rna;
  } 
};

module.exports = DnaTranscriber;
